core/src/main/java/hudson/model/FreeStyleProject.java

```diff
@@ -72,5 +74,25 @@ public String getDisplayName() {
         public FreeStyleProject newInstance(ItemGroup parent, String name) {
             return new FreeStyleProject(parent,name);
         }
+
+        @Override
+        public String getDescription() {
+            return Messages.FreeStyleProject_Description();
+        }
+
+        @Override
+        public String getCategoryId() {
+            return StandaloneProjectsCategory.ID;
+        }
+
+        @Override
+        public String getIconFilePathPattern() {
+            if (Jenkins.RESOURCE_PATH.isEmpty()) {
+                return "images/:size/freestyleproject.png";
+            } else {
+                return Jenkins.RESOURCE_PATH.substring(1) + "/images/:size/freestyleproject.png";
```

Comment:
@jglick 

> did not really explain anything, but the if switch seems like overkill.

I think the code is clear to understand what does this method.

> `Jenkins.RESOURCE_PATH + "/images/:size/freestyleproject.png").replaceFirst("^/", "");`

Different styles, only that.